### ISSUE TITLE
Does Mask-Attention really work?

### ISSUE THREAD

**Issue by Jummmmmp** [frank-xwang/InstanceDiffusion#16]
I set `return_att_masks = True` in _python inference.py_ and use the following code to run your model:`python inference.py   --num_images 8   --output OUTPUT/   --input_json demos/demo_corgi_kitchen.json   --ckpt pretrained/instancediffusion_sd15.pth   --test_config configs/test_box.yaml   --guidance_scale 7.5   --alpha 0.8   --seed 0   --mis 0.36   --cascade_strength 0.4`. In contrast, I deleted all the "mask" value in _demo_corgi_kitchen.json_ and tried again, but the result still remain the same. So I wonder does Mask-Attention really work? Or is there any proper way to use Mask-Attention? Wish your early reply, thanks!

**Comment by frank-xwang** [comment]
Hi, for increasing inference speed and reduce the GPU memory usage, we've implemented Flash Attention in our inference codes. Currently, Flash Attention does not support attention masking. Furthermore, our ablation study in the paper, detailed in the attached screenshot, indicates that the exclusion of the attention mask results in only a small drop in performance.

![image](https://github.com/frank-xwang/InstanceDiffusion/assets/58996472/1a47d62f-b9c9-4612-869f-8217e2ba6cc2)

As a result, attention masking has been disabled by default to prioritize faster inference times. If you wish to enable mask-attention, you can do so by setting `efficient_attention` to `false` in the configuration file located at `configs/test_box.yaml`.

Hope it helps.

**Comment by Jummmmmp** [comment]
I try to change _test_box.yaml_ to _test_mask.yaml_ and then the problem solved! I can test mask-attention now. It's so kind of you. Thanks!
